module/Dashboard/src/Dashboard/Model/Analyzer/NumberAnalyzer.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * @author pdziok
+ */
+namespace Dashboard\Model\Analyzer;
+
+use Dashboard\Model\Widget\NumberWidget;
+
+class NumberAnalyzer extends AbstractAnalyzer
+{
+    private $archive;
+    private $samples;
+
+    public function __construct(NumberWidget $widget)
+    {
+        parent::__construct($widget);
+
+        $cfg = $widget->getParam('analyze');
+
+        $secondsToAnalyze = $this->intervalToSeconds($cfg['last']);
+        $frequency = isset($cfg['each'])
+            ? $this->intervalToSeconds($cfg['each'])
+            : 60;
+        $this->samples = ceil($secondsToAnalyze / $frequency);
+    }
+
+    public function analyze()
```

Comment:
This method is duplicated between GraphAnalyzer and NumberAnalyzer - please refactor.